### DISCUSSION TITLE
Custom Groups/Roles in Apostrophe 3?

### DISCUSSION THREAD

**Discussion by mfreeman-xtivia** [apostrophecms/apostrophe#3284] [Q&A] [answered]
Is it possible to define/use custom groups in A3?

There is documentation that describes how to do this in A2, but i do not see any similar documentation for A2

**Comment by boutell** [comment, marked as answer]
Hi @mfreeman-xtivia, in A3 the standard permissions roles are "guest," "contributor," "editor" and "admin." Since all documents start out as drafts and contributors can work with drafts and submit them for review but cannot publish, this provides a simple workflow that replaces most common use cases for custom groups, up to a certain scale.

That being said, we understand there are larger scales where custom groups will still make sense, especially in the enterprise, which is why you can find Custom User Groups on our product board as an upcoming Apostrophe Assembly feature:

https://portal.productboard.com/apostrophecms/1-product-roadmap/tabs/2-planned

**Reply by MartinNevlaha** [reply]
Hello, and is possible to remove for example two roles contributor and guest ?

**Reply by abea** [reply]
@MartinNevlaha In your project you could add a `modules/@apostrophecms/user/index.js` file and override the `role` field there to remove those as choices.

```
module.exports = {
  fields: {
    add: {
        role: {
          type: 'role',
          choices: [
            {
              label: 'apostrophe:editor',
              value: 'editor'
            },
            {
              label: 'apostrophe:admin',
              value: 'admin'
            }
          ],
          def: 'editor',
          required: true
        }
    }
  }
}
```

**Reply by MartinNevlaha** [reply]
@abea thanks

**Comment by mfreeman-xtivia** [comment]
So this will never be available in the non-Assembly, OSS version of Apostrophe 3??

**Reply by abea** [reply]
"Never" might be too strong given how software and business models evolve but, for the foreseeable future, core Apostrophe 3 is replacing custom groups with those four user roles. We may explore more a la carte options for advanced features like this at some point, but certain features like this are part of the commercialization strategy that will support Apostrophe as OSS in the long term.

The A2 group permissions configuration was often more confusion than it was worth for the great majority of smaller-scale projects, especially when content reviews came into play. The pre-defined user roles are meant to satisfy those majority cases.

**Comment by boutell** [comment]
That doesn't sound right, can we see the complete module code?

On Sun, Sep 12, 2021 at 5:40 AM MartinNevlaha ***@***.***>
wrote:

> @abea <https://github.com/abea> hmmm this doesn't work, i get an error
> when app starts : user: fields is not a valid top level property for an
> Apostrophe 3.x module. Make sure you nest regular module options in the new
> "options" property.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/apostrophecms/apostrophe/discussions/3284#discussioncomment-1312035>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAH27ID36HWTV755A23F73UBRYPPANCNFSM5B22EXRA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 

THOMAS BOUTELL | CHIEF TECHNOLOGY OFFICER
APOSTROPHECMS | apostrophecms.com | he/him/his

**Reply by MartinNevlaha** [reply]
sorry that was my mistake, I put user folder with index.js right into modules no into modules/@apostrophecms. Now it works properly.
